notebooks/nearest_neighbors.livemd

```diff
@@ -0,0 +1,255 @@
+<!-- livebook:{"file_entries":[{"name":"knn_gyms.png","type":"attachment"}]} -->
+
+# Nearest Neighbors
```

Comment:
Since we already have a notebook on kNN, what if we call this "Approximate nearest neighbours"?

Reply:
Hrm, we also talk about KDTree here, which is not approximate. But KDTree is `k=1` right? So I think this notebook is good as is. :)

Reply:
This notebook also explains KDTrees which aren't approximate NN. I know the current name somehow collides with the notebook about brute-force implementation. I'll think about a new name.

Reply:
$K$ in KD stand for the number of dimensions of data. In the example $K = 7$.